features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/TestConsole/TestConsole.jsx

```diff
@@ -270,103 +258,79 @@ class TestConsole extends React.Component {
     }
 
     /**
-     * Set Production Access Token
+     * Load the swagger file of the given environment
      * @memberof ApiConsole
      */
-    setProductionAccessToken(productionAccessToken) {
-        this.setState({ productionAccessToken });
-    }
-
-    /**
-     * Set Sandbox Access Token
-     * @memberof ApiConsole
-     */
-    setSandboxAccessToken(sandboxAccessToken) {
-        this.setState({ sandboxAccessToken });
-    }
-
-    /**
-     * Set Username
-     * @memberof ApiConsole
-     */
-    setUsername(username) {
-        this.setState({ username });
-    }
-
-    /**
-     * Set Password
-     * @memberof ApiConsole
-     */
-    setPassword(password) {
-        this.setState({ password });
-    }
+    updateSwagger(environment) {
+        let urls;
+        const {
+            api,
+        } = this.state;
 
-    /**
-     * Set Password
-     * @memberof ApiConsole
-     */
-    setSelectedKeyType(selectedKeyType, isUpdateToken) {
-        if (isUpdateToken) {
-            this.setState({ selectedKeyType }, this.updateAccessToken);
-        } else {
-            this.setState({ selectedKeyType });
-        }
+        const promiseSwagger = Api.getSwaggerByAPIIdAndEnvironment(api.id, environment);
+        const settingPromise = Api.getSettings();
+        settingPromise
+            .then((settingsNew) => {
+                if (settingsNew.environment) {
+                    urls = settingsNew.environment.map((envo) => {
+                        const env = {
+                            name: envo.name,
+                            endpoints: {
+                                http: envo.endpoints.http + api.context + '/' + api.version,
+                                https: envo.endpoints.https + api.context + '/' + api.version,
+                            },
+                        };
+                        return env;
+                    });
+                }
+                this.setState({
+                    settings: urls,
+                });
+            });
+        promiseSwagger
+            .then((swaggerResponse) => {
+                this.setState({ swagger: swaggerResponse.obj });
+            });
     }
 
     /**
-     * Set Password
-     * @memberof ApiConsole
-     */
-    setKeys(keys) {
-        this.setState({ keys });
+    * Generate Internal-Token
+    */
+    generateKey() {
+        let key;
+        const {
+            api,
+        } = this.state;
+        const promisedAPI = Api.generateInternalKey(api.id);
+        promisedAPI
+            .then((apiResponse) => {
+                key = apiResponse.obj.apikey;
+                this.setState({
+                    key,
+                    showToken: false,
+                });
+            });
+        return key;
```

Comment:
This will always return `undefined` since we set the key inside an async function, `generateKey ` should return a promise then